// @solutions/server/src/routes/auth.ts
import { ensureAuth } from "../middlewares/ensureAuth";
import { Router } from "express";
import { z } from "zod";
import bcrypt from "bcryptjs";
import jwt from "jsonwebtoken";
import { prisma } from "../lib/prisma";

export const authRouter = Router();

const RegisterSchema = z.object({
  orgName: z.string().min(2),
  name: z.string().min(2),
  email: z.string().email(),
  password: z.string().min(6),
});

authRouter.post("/auth/register", async (req, res) => {
  const parsed = RegisterSchema.safeParse(req.body);
  if (!parsed.success) return res.status(400).json({ error: "invalid_body", issues: parsed.error.issues });

  const { orgName, name, email, password } = parsed.data;

  const exists = await prisma.user.findUnique({ where: { email } });
  if (exists) return res.status(409).json({ error: "email_in_use" });

  const org = await prisma.organization.create({ data: { name: orgName } });
  const user = await prisma.user.create({
    data: { name, email, password: await bcrypt.hash(password, 10) },
  });
  await prisma.membership.create({
    data: { orgId: org.id, userId: user.id, role: "owner" },
  });

  const secret = process.env.JWT_SECRET || "change_me";
  const token = jwt.sign({ userId: user.id, orgId: org.id, role: "owner" }, secret, { expiresIn: "7d" });
  res.json({ token, orgId: org.id, user: { id: user.id, name: user.name, email: user.email } });
});

const LoginSchema = z.object({
  email: z.string().email(),
  password: z.string().min(1),
  orgId: z.string().optional(),
});

authRouter.post("/auth/login", async (req, res) => {
  const parsed = LoginSchema.safeParse(req.body);
  if (!parsed.success) return res.status(400).json({ error: "invalid_body" });

  const { email, password } = parsed.data;
  const user = await prisma.user.findUnique({ where: { email }, include: { memberships: true } });
  if (!user) return res.status(401).json({ error: "bad_credentials" });

  const ok = await bcrypt.compare(password, user.password);
  if (!ok) return res.status(401).json({ error: "bad_credentials" });

  const membership = user.memberships[0];
  if (!membership) return res.status(403).json({ error: "no_org" });

  const secret = process.env.JWT_SECRET || "change_me";
  const token = jwt.sign({ userId: user.id, orgId: membership.orgId, role: membership.role }, secret, { expiresIn: "7d" });

  res.json({
    token,
    orgId: membership.orgId,
    role: membership.role,
    user: { id: user.id, name: user.name, email: user.email },
  });
});
authRouter.get("/me", ensureAuth, async (req, res) => {
const authHeader = req.headers.authorization;

  if (!authHeader) {
    return res.status(401).json({ error: "missing_token" });
  }

  const [, token] = authHeader.split(" ");

  try {
    const secret = process.env.JWT_SECRET || "change_me";
    const decoded = jwt.verify(token, secret) as {
      userId: string;
      orgId: string;
      role: string;
    };

    const user = await prisma.user.findUnique({
      where: { id: decoded.userId },
      select: {
        id: true,
        name: true,
        email: true,
      },
    });

    if (!user) {
      return res.status(404).json({ error: "user_not_found" });
    }

    res.json({
      user,
      orgId: decoded.orgId,
      role: decoded.role,
    });
  } catch (err) {
    return res.status(401).json({ error: "invalid_token" });
  }
});
